src/NServiceBus.Core/Receiving/ReceiveComponent.cs

```diff
@@ -151,7 +151,8 @@ partial class ReceiveComponent
                 return;
             }
 
-            var mainPump = CreateReceiver(consecutiveFailuresConfiguration, transportInfrastructure.Receivers[MainReceiverId]);
+            var mainReceiver = transportInfrastructure.Receivers[MainReceiverId];
+            var mainPump = CreateReceiver(consecutiveFailuresConfiguration, mainReceiver);
```

Comment:
We can probably revert this change as it's not needed. It's not hurting anything either but it will be noise in the history.